File: miscellaneous/roman_to_int.py
'''
LeetCode - problem 13
'''

    def romanToInt(self, s):
        ns = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000,
        }
        r=0
        ps=0
        for i in range(len(s)):
            r+=ns[s[i]]
            if i>0 and ns[s[i-1]]<ns[s[i]]:
                r-=2*ns[s[i-1]]
        return r

File: miscellaneous/roman_to_int.cpp
//LeetCode - problem 13

int romanToInt(string s) {
        unordered_map<char,int>m;
        int ans=0;
        m['I']=1;
        m['V']=5;
        m['X']=10;
        m['L']=50;
        m['C']=100;
        m['D']=500;
        m['M']=1000;
        int i=0;
            while(s[i]!='\0'){
                if(m[s[i]]<m[s[i+1]]){
                    ans=ans+m[s[i+1]]-m[s[i]];
                    i=i+2;
                }
                else{
                    ans=ans+m[s[i]];
                    i++;
                    
                }     
            }
                return ans;     
            
        
    }
